src/OpenTelemetry.Contrib.Instrumentation.AWS/Implementation/AWSTracingPipelineHandler.cs

```diff
@@ -0,0 +1,270 @@
+            catch (Exception ex)
+            {
+                if (activity != null)
+                {
+                    this.ProcessException(activity, ex);
+                }
+
+                throw;
+            }
+            finally
+            {
+                if (activity != null)
+                {
+                    this.ProcessEndRequest(executionContext, activity);
+                }
+            }
+
+            return ret;
+        }
+
+        private Activity ProcessBeginRequest(IExecutionContext executionContext)
+        {
+            Activity activity = null;
+
+            var requestContext = executionContext.RequestContext;
+            var service = this.GetAWSServiceName(requestContext);
+            var operation = this.GetAWSOperationName(requestContext);
+
+            activity = AWSSDKActivitySource.StartActivity(service + "." + operation, ActivityKind.Client);
+
+            if (activity == null)
+            {
+                return null;
+            }
+
+            if (this.options.SuppressDownstreamInstrumentation)
+            {
+                SuppressInstrumentationScope.Enter();
+            }
+
+            if (activity.IsAllDataRequested)
+            {
+                activity.SetTag(AWSSemanticConventions.AttributeAWSServiceName, service);
+                activity.SetTag(AWSSemanticConventions.AttributeAWSOperationName, operation);
+                var client = executionContext.RequestContext.ClientConfig;
+                if (client != null)
+                {
+                    var region = client.RegionEndpoint?.SystemName;
+                    activity.SetTag(AWSSemanticConventions.AttributeAWSRegion, region ?? AWSSDKUtils.DetermineRegion(client.ServiceURL));
+                }
+
+                this.AddRequestSpecificInformation(activity, requestContext, service);
+            }
+
+            AwsPropagator.Inject(new PropagationContext(activity.Context, Baggage.Current), requestContext.Request.Headers, Setter);
+
+            return activity;
+        }
+
+        private void ProcessEndRequest(IExecutionContext executionContext, Activity activity)
+        {
+            var responseContext = executionContext.ResponseContext;
+            var requestContext = executionContext.RequestContext;
+
+            if (activity.IsAllDataRequested)
+            {
+                if (Utils.GetTagValue(activity, AWSSemanticConventions.AttributeAWSRequestId) == null)
+                {
+                    activity.SetTag(AWSSemanticConventions.AttributeAWSRequestId, this.FetchRequestId(requestContext, responseContext));
+                }
+
+                var httpResponse = responseContext.HttpResponse;
+                if (httpResponse != null)
+                {
+                    int statusCode = (int)httpResponse.StatusCode;
+
+                    var status = activity.GetStatus();
+
+                    if (!status.Equals(Status.Error) && statusCode >= 100 && statusCode <= 399)
```

Comment:
status is by default `Unset`.. Not sure why setting it to Unset here ?

Reply:
Hmm. Was it always defaulting to `Unset`? I added this code here probably because it wasn't by default but can't seem to exactly remember.
Since `Unset` is default now, we can remove this block for checking.

Reply:
Done. Removed this code block.